nengo_bones/scripts/generate_bones.py

```diff
@@ -131,128 +90,87 @@ def ci_scripts(ctx):
     config = ctx.obj["config"]
     for params in config["ci_scripts"]:
         script_name = params.pop("template")
-        template = ctx.obj["env"].get_template(script_name + ".sh.template")
-        output_name = params.pop("output_name", script_name)
-        path = os.path.join(ctx.obj["output_dir"], output_name + ".sh")
-
-        with open(path, "w") as f:
-            f.write(template.render(
-                pkg_name=config["pkg_name"], repo_name=config["repo_name"],
-                version=nengo_bones.__version__, **params))
-
-        # Mark CI script as executable
-        st = os.stat(path)
-        os.chmod(path, st.st_mode | stat.S_IEXEC)
+        output_file = params.pop("output_name", script_name)
+        BonesTemplate(script_name + ".sh").render_to_file(
+            ctx.obj["output_dir"],
+            output_name=output_file + ".sh",
+            version=__version__,
+            pkg_name=config["pkg_name"],
+            repo_name=config["repo_name"],
+            **params,
```

Comment:
This logic could go in `flatten_config` (renamed to e.g. `get_render_data`) along with the other template-specific render argument logic.

Reply:
CI scripts are a bit more complicated since their output filename doesn't correspond to a section name ... I think more refactoring would have to happen to get CI scripts working the same way, so for now I'd vote to leave this as is.